simtools/data_model/validate_data.py

```diff
@@ -293,6 +294,33 @@ def _get_reference_unit(self, column_name):
 
         return u.Unit(reference_unit)
 
+    def _check_data_type(self, col, column_name):
```

Comment:
the remaining functions usually return the final corrected value. I suggest doing the same for this function.

Reply:
The difference with this function is that this is an 'either-the-type-is-correct-or-raise-error-function'.

For the other ones, we can convert (e.g., if a value is given in "km", and we convert it to "m".

Reply:
Added a DEBUG statement.

Reply:
sounds good!